test/controllers/session_controller_test.exs

```diff
@@ -0,0 +1,38 @@
+defmodule ResearchResource.SessionControllerTest do
+  use ResearchResource.ConnCase
+
+  setup %{conn: conn} = config do
+    if username = config[:login_as] do
```

Comment:
You could just write `config.login_as` instead of `config[:login_as]` to be more consistent with how you are getting atom properties from maps